src/sqlfluff/rules/L052.py

```diff
@@ -93,10 +93,32 @@ def _handle_trailing_inline_comments(context, anchor_segment):
 
         return anchor_segment
 
+    @staticmethod
+    def _is_one_line_statement(context, segment):
+        """Check if the statement containing the provided segment is one line."""
+        # Find statement segment containing the current segment.
+        statement_segment = next(
+            (
+                s
+                for s in context.parent_stack[0].path_to(segment)
+                if s.is_type("statement")
+            ),
+            None,
+        )
+        if statement_segment is None:  # pragma: no cover
+            # If we can't find a parent statement segment then don't try anything special.
+            return False
+
+        if not list(statement_segment.recursive_crawl("newline")):
```

Comment:
Performance nitpick:  Since `recursive_crawl()` is a generator, I think you could change this from `list()` to `any()`, then it would stop early if it finds a newline.